### PR TITLE
feat: load open id token for kyma auth

### LINKED ISSUE #169: Secure lambda calls to Kyma / SPA-1415
To do:
- Talk to Johannes and Pascal and determine what we need to to.
- Create tickets to do it (use epic/kymaauth) or add subtasks in this ticket

Original ticket - SPA-1415:

To perform a authenticated call to Kyma from SPA, the developer needs to be able to pass to kyma an oauth id_token retrieved from commerce platform during the user login. 

This id_token is essential to allow Kyma to verify the lambda caller identity with the commerce platform acting as OpenId IdP (Identity Provider)

Current SPA Login code based on Oauth only retrieve an access token from the commerce platform. It should also ask the commerce platform for a Id token and store this additional token in the user object for later use by other part of SPA code requiring to make calls to Kyma.

For CX Live a workaround was put together: https://wiki.hybris.com/display/~johannes.engelke@hybris.com/Ugly+Workaround+to+fetch+id_token+in+SPA

### PR DESCRIPTION
**Pull request by znikola**
closes #169

### CODE REVIEW

File: projects/core/src/auth/models/token-types.model.ts [hunk 1]
```diff
@@ -15,4 +15,12 @@ export interface ClientToken {
   scope: string;
 }
 
-export type AuthenticationToken = UserToken | ClientToken;
+export interface OpenIdToken {
```
Comment (hackergil) [inline projects/core/src/auth/models/token-types.model.ts]: Can we simplify this file to maybe two interfaces or extending from a common one?

File: projects/core/src/auth/services/open-id-token/open-id-token.service.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+export class OpenIdAuthenticationTokenService {
+  constructor(private config: AuthConfig, private http: HttpClient) {}
+
+  loadOpenIdAuthenticationToken(
+    username: string,
+    password: string
+  ): Observable<OpenIdToken> {
+    const url = this.getOAuthEndpoint();
+    const params = new HttpParams()
+      .set('client_id', this.config.authentication.kyma_client_id)
```
Comment (hackergil) [inline projects/core/src/auth/services/open-id-token/open-id-token.service.ts]: Shall we encodeuri these in a similar way than client-authentication-token.service ?

**Comment by znikola** [comment]
> My understanding is that kyma auth token is just another auth token. Is there something that makes it different or special? Can’t we just treat all of our tokens in a uniform way (using the same interface) across the board?

The token needed for kyma is called OpenID token, which kyma needs. Since the kyma is a new OAuth client (different than the one we are currently using) we need to store the whole payload for kyma to use.

We could possible combine it with `UserToken`, but we'd have to rename the interface in that case.

File: projects/core/src/auth/models/token-types.model.ts (hunk 1 above)
Comment (znikola) [reply projects/core/src/auth/models/token-types.model.ts]: Good catch, we could

File: projects/core/src/auth/services/open-id-token/open-id-token.service.ts (hunk 2 above)
Comment (znikola) [reply projects/core/src/auth/services/open-id-token/open-id-token.service.ts]: Done

**Pull request merged**